### DISCUSSION TITLE
LND Rest connection refused to a remote BTCPayServer

### DISCUSSION THREAD

**Discussion by dme1sc** [lightningnetwork/lnd#8807] [Troubleshooting]
I have LND on a start9 personal server. I have a BTCPayServer on a remote hosted clearnet server.
I'm trying to connect btcpay to LND using the lnd-rest connection.

BTCPay has the connection built as 
type=lnd-rest;server=http://address.onion:8080;macaroon=XXXXXXhexidecimalformat;allowinsecure=true;
If I do it this way, I get:
The lightning node did not reply in a timely manner.
The LND logs show:
[ERR] CMGR: Can't accept connection: unable to accept connection from x.x.x.x:51540: read tcp x.x.x.y:9735->x.x.x.x:51540: i/o timeout

If I change the onion address to an https clearnet domain name with a reverse proxy in this format:
type=lnd-rest;server=https://address.com:8080;macaroon=XXXXXXhexidecimalformat;
I get:
Error while connecting to the API: Connection refused (lnd.domainname.com:8080) (Connection refused)
I found that my reverse proxy was forwarding to port 80, I switched to 8080 in the proxy, and also removed 8080 from the URL:
type=lnd-rest;server=https://address.com;macaroon=XXXXXXhexidecimalformat;
and now just get a timeout consistently with The lightning node did not reply in a timely manner.

If I put the 8080 back in the URL I get:
Error while connecting to the API: Connection refused  (lnd.domainname.com:8080)  (Connection refused)

But if I test the onion address to 8080 (http://address.onion:8080), I get a JSON page.
It seems my reverse proxy is misconfigured but I'm not sure what else to check as the reverse proxy in this format works for other services.

**Comment by guggero** [comment]
What's your `lnd` config? Do you have `restlisten=0.0.0.0:8080` set?

**Comment by dme1sc** [comment]
yes, confirmed that it is 8080.

**Reply by dme1sc** [reply]
I was just told by someone that btcpay may not support what I'm trying to do.
BTCPay on remote host using reverse proxy to a remote TOR server using REST in the connection string.  It does seem a bit complicated what I am trying to do.
